census/tflearn/README.md

```diff
@@ -0,0 +1,111 @@
+# Census: TF.Learn Distributed Training and Prediction
+
+This sample walks you through training and prediction, including distributed training, and hyperparameter tuning.
+
+## Local Training
+
+### Getting Data
+
+Throughout we will use a census dataset for training. For local training, you can read them directly from Google Cloud Storage or if you have an unreliable network, can download them to your local file system.
+
+```
+TRAIN_FILE=gs://tf-ml-workshop/widendeep/adult.data
+EVAL_FILE=gs://tf-ml-workshop/widendeep/adult.test
+```
+
+To use locally simply copy them down with gsutil:
+```
+gsutil cp gs://tf-ml-workshop/widendeep/* data/
+```
+
+And use the local locations whereever the environment variables above are used.
+
+
+### Installing Dependencies
+
+The local training environment for Cloud ML uses your local python installation. So you will need to have Python 2.7 and install TensorFlow following the instructions [here](https://www.tensorflow.org/versions/r1.0/get_started/os_setup). (Note please ensure that you install `tensorflow>=1.0.0-rc2`)
+
+### Running Training
+
+Then you can run training with the following command:
+
+```
+gcloud beta ml local train \
+    --module-name trainer.task \
+    --package-path trainer/ \
+    -- \
+    --train-file $TRAIN_FILE \
+    --eval-file $EVAL_FILE \
+    --train-steps 1000 \
+    --job_dir output
+```
+
+You should see very verbose output, ending with a line about saving your model.
+
+To ensure that your model works with Cloud MLs distributed execution environment you can add the `--distributed` flag (before the `--` that seperates user args from CLI args). You will see output from a number of different processes which are communicating through gRPC!
+
+## Local Prediction
```

Comment:
Is there a way we can decouple the local prediction tool from the Cloud ML SDK... It's really helpful. I literally would not have figured out my issue with prediction had I not had it around. Ideally it should run one of the tensorflow serving docker containers locally? After GA we need to have a much more in depth discussion of the local emulator, since my experiences indicate it needs to be beefed up.